### DISCUSSION TITLE
Please support VS Code syntax coloring: printing as-displayed with our favourite theme and 0 setup

### DISCUSSION THREAD

**Discussion by maximedrouhin** [PDConSec/vsc-print#252] [Ideas]
Create a webview for Print settings with an integrated hot preview for syntax highlighting colour schemes.

> I think the experience would be massively improved if you could manage using the VS Code syntax coloring when printing.
>
> This way:
> 1. The code is printed as-displayed with the syntax coloring the user chose and is used to, instead of having to choose between unfamiliar syntax colorings. Even if the user uses a dark theme, many syntax colorings have their dark counterparts (e.g. Solarized Dark -> Solarized Light), that the user is likely to like. In other words, things can now work with 0 setup since the user already knows and likes the syntax coloring theme used for printing.
> 2. The user can easily test themes and preview them by using the "Preferences: Color Theme" Command Palette option. Currently, it's very hard to test themes, you either use the [highlight.js demo](https://highlightjs.org/static/demo/) website, which doesn't enable to preview your own file, or you have to click the Print button over and over again, changing the settings each time.

**Comment by PeterWone** [comment]
For your first item please read this https://github.com/PDConSec/vsc-print/discussions/251

Mapping of editor schemes to highlightjs schemes might allow partial automation but 

* it would be inconsistent in success and that would produce an endless stream of unhappy users
* someone would have to maintain the mappings forever

As regards hot preview of a scheme, that's not out of the question so I'm going to move this to the ideas discussion. I'll also create another item in Ideas discussion for supporting the mapping of scheme names. That's not a promise to implement, just to consider the practicalities of building tooling to make it less onerous for a volunteer to maintain mappings.

**Comment by maximedrouhin** [comment]
Thanks a lot for your answer! And sorry I missed #251...

**Comment by PeterWone** [comment]
Think nothing of it. We get this question about once a year and you prompted me to add a comprehensive consideration to Q+A.
